src/stonkgs/api/example.py

```diff
@@ -0,0 +1,105 @@
+# -*- coding: utf-8 -*-
+
+"""Full example of how to use a fine tuned model.
+
+Run with: ``python -m stonkgs.api.example``
+"""
+
+import pickle
+import time
+
+import click
+import pandas as pd
+import pystow
+import torch
+from datasets import Dataset
+from tqdm import tqdm
+from transformers.trainer_utils import PredictionOutput
+
+from stonkgs import STonKGsForSequenceClassification, preprocess_df_for_embeddings
+from stonkgs.api.constants import ensure_embeddings, ensure_species, ensure_walks
+
+
+def _onrows(source_df: pd.DataFrame, model):
+    click.echo("Processing df for embeddings")
+    t = time.time()
+    preprocessed_df = preprocess_df_for_embeddings(
+        df=source_df,
+        embedding_name_to_vector_path=ensure_embeddings(),
+        embedding_name_to_random_walk_path=ensure_walks(),
+    )[["input_ids", "attention_mask", "token_type_ids"]]
+    click.echo(f"done processing df for embeddings after {time.time() - t:.2f} seconds")
+
+    dataset = Dataset.from_pandas(preprocessed_df)
+    dataset.set_format("torch")
+
+    # Initialize Softmax to process the logits predictions later on
+    softmax = torch.nn.Softmax(dim=1)
+
+    # Save both the raw prediction results (as a pickle) as well as the processed probabilities (in a dataframe)
+    raw_results = []
+    probabilities = []
+    for idx, _ in tqdm(preprocessed_df.iterrows(), desc="Inferring"):
+        # Process each row at once
+        data_entry = {
+            key: torch.tensor([value]) for key, value in dict(preprocessed_df.iloc[idx]).items()
+        }
+
+        prediction_output: PredictionOutput = model(**data_entry, return_dict=True)
+        probabilities.append(softmax(prediction_output.logits)[0].tolist())
```

Comment:
how am I supposed to interpret the results coming out of this? These numbers aren't between 0 and 1 so they're not really probabilities:

<img width="922" alt="Screen Shot 2021-08-17 at 09 05 19" src="https://user-images.githubusercontent.com/5069736/129679590-fe506362-d8ad-4777-91b9-6796c040b9a8.png">